In [147]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', 100)
%matplotlib inline

In [148]:
df = pd.read_csv('vectorized_model_data.csv')

In [149]:
recipe_data = pd.read_csv('recipe_data.csv')

In [150]:
df.shape

(39774, 917)

In [151]:
recipe_data.shape

(39774, 6)

In [152]:
recipe_total = 39774

In [153]:
recipe_data.groupby('cuisine')['cuisine'].count()

cuisine
brazilian        467
british          804
cajun_creole    1546
chinese         2673
filipino         755
french          2646
greek           1175
indian          3003
irish            667
italian         7838
jamaican         526
japanese        1423
korean           830
mexican         6438
moroccan         821
russian          489
southern_us     4320
spanish          989
thai            1539
vietnamese       825
Name: cuisine, dtype: int64

In [154]:
cuisine_counts = {
    'brazilian':467,
    'british':804,
    'cajun_creole':546,
    'chinese':2673,
    'filipino':755,
    'french':2646,
    'greek':1175,
    'indian':3003,
    'irish':667,
    'italian':7838,
    'jamaican':526,
    'japanese':1423,
    'korean':830,
    'mexican':6438,
    'moroccan':821,
    'russian':489,
    'southern_us':4320,
    'spanish':989,
    'thai':1539,
    'vietnamese':825   
}

In [155]:
cuisine_percentages = {}
for key in cuisine_counts:
    cuisine_percentages[key] = cuisine_counts[key] / recipe_total

In [156]:
cuisine_percentages

{'brazilian': 0.011741338562880274,
 'british': 0.020214210288127924,
 'cajun_creole': 0.013727560718057022,
 'chinese': 0.06720470659224619,
 'filipino': 0.018982249710866397,
 'french': 0.0665258711721225,
 'greek': 0.02954191180167949,
 'indian': 0.07550158394931362,
 'irish': 0.01676974908231508,
 'italian': 0.19706340825665009,
 'jamaican': 0.013224719666113542,
 'japanese': 0.03577714084577865,
 'korean': 0.020867903655654448,
 'mexican': 0.16186453462060643,
 'moroccan': 0.020641625182279882,
 'russian': 0.012294463720018103,
 'southern_us': 0.10861366721979182,
 'spanish': 0.024865490018605118,
 'thai': 0.03869361894705084,
 'vietnamese': 0.020742193392668577}

In [157]:
df['cuisine'] = recipe_data['cuisine']

In [158]:
ingredient_counts = df.groupby('cuisine').sum().reset_index()

In [159]:
#' mozzarella' and &half didn't work, so drop them
ingredient_counts.drop([' mozzarella', '& half'], axis=1, inplace=True)

In [160]:
ingredient_counts.head(20)

cuisine  active yeast  adobo sauce  agave nectar  alfredo sauce  \
0      brazilian             3            0             4              0   
1        british            13            0             0              0   
2   cajun_creole            36            0             0              5   
3        chinese            12            0             4              0   
4       filipino             7            1             0              0   
5         french            40            0             1              1   
6          greek             6            0             2              0   
7         indian            34            1             4              0   
8          irish             1            1             0              0   
9        italian           128            1             2             67   
10      jamaican             2            1             0              0   
11      japanese             2            0            11              0   
12        korean             0            0             6              0   
13       mexican            17           85            28              1   
14      moroccan             8            0             0              0   
15       russian            35            0             2              0   
16   southern_us            29            4             7              0   
17       spanish             4            0             1              0   
18          thai             1            0             5              0   
19    vietnamese             1            0             6              0   

    allspice  almond  almond extract  almond flour  almond milk  amchur  \
0          1      24               0             0           13       0   
1         27      44              11             1            1       0   
2         18      12               2             1            0       0   
3          0      46               8             5            0       0   
4          1       2               1             0            0       0   
5         38     176              44            13            3       0   
6         32      29               3             0            0       0   
7         24     124               1             4            8      69   
8         20      13               7             0            0       0   
9         12     266              56            15            6       0   
10       195       3               1             1            0       0   
11         1      13               2             1            1      17   
12         0       1               0             0            0       0   
13        66      83               6             4            3       0   
14        44     111               1             0            0       0   
15         8      26               5             0            0       0   
16        59      70              23            10            5       1   
17         4      94               2             3            1       0   
18         1       8               0             1            0       0   
19         4       4               0             1            0       0   

    american cheese  anaheim chile  ancho  ancho chile pepper  anchovy  \
0                 0              0      1                   0        0   
1                 0              0      0                   0        1   
2                 2              0      2                   0        5   
3                 0              0      0                   0        0   
4                 1              1      0                   0        0   
5                 0              0      1                   0       71   
6                 0              0      0                   0        5   
7                 0              1      1                   0        0   
8                 0              0      0                   0        0   
9                 5              0      1                   0      202   
10                0   

In [161]:
ic2 = ingredient_counts.set_index('cuisine').T

In [162]:
ic2.head()

cuisine        brazilian  british  cajun_creole  chinese  filipino  french  \
active yeast           3       13            36       12         7      40   
adobo sauce            0        0             0        0         1       0   
agave nectar           4        0             0        4         0       1   
alfredo sauce          0        0             5        0         0       1   
allspice               1       27            18        0         1      38   

cuisine        greek  indian  irish  italian  jamaican  japanese  korean  \
active yeast       6      34      1      128         2         2       0   
adobo sauce        0       1      1        1         1         0       0   
agave nectar       2       4      0        2         0        11       6   
alfredo sauce      0       0      0       67         0         0       0   
allspice          32      24     20       12       195         1       0   

cuisine        mexican  moroccan  russian  southern_us  spanish  thai  \
active yeast        17         8       35           29        4     1   
adobo sauce         85         0        0            4        0     0   
agave nectar        28         0        2            7        1     5   
alfredo sauce        1         0        0            0        0     0   
allspice            66        44        8           59        4     1   

cuisine        vietnamese  
active yeast            1  
adobo sauce             0  
agave nectar            6  
alfredo sauce           0  
allspice                4

In [163]:
raw_counts = ic2.reset_index().rename(columns={'index':'ingredient'}) 

In [164]:
raw_counts.head(40)

cuisine          ingredient  brazilian  british  cajun_creole  chinese  \
0              active yeast          3       13            36       12   
1               adobo sauce          0        0             0        0   
2              agave nectar          4        0             0        4   
3             alfredo sauce          0        0             5        0   
4                  allspice          1       27            18        0   
5                    almond         24       44            12       46   
6            almond extract          0       11             2        8   
7              almond flour          0        1             1        5   
8               almond milk         13        1             0        0   
9                    amchur          0        0             0        0   
10          american cheese          0        0             2        0   
11            anaheim chile          0        0             0        0   
12                    ancho          1        0             2        0   
13       ancho chile pepper          0        0             0        0   
14                  anchovy          0        1             5        0   
15            anchovy paste          0        0             3        0   
16        andouille sausage          1        0           228        0   
17               angel hair          0        0             4        4   
18                    anise          2        1             0      111   
19               anise seed          1        0             0        1   
20                    apple         10       30            15       26   
21              apple cider          1        1             6       23   
22              apple juice          1        0             0        1   
23                  apricot          1       14             0        4   
24             arborio rice          0        0             3        1   
25                artichoke          0        1            10        0   
26          artichoke heart          0        0             4        0   
27                  arugula          0        3             2        1   
28                asafetida          0        0             0        0   
29               asafoetida          0        0             0        0   
30                   asiago          0        0             1        0   
31                 asparagu          0        1             6       20   
32           asparagu spear          0        0             1        3   
33                  avocado         10        0             4        9   
34                    bacon         30       25            62       26   
35           bacon dripping          0        2             7        0   
36                 baguette          2        0            11        1   
37                   baking         19      214            62      100   
38            baking potato          0       13             4        1   
39              baking soda          5       73            23       46   

cuisine  filipino  french  greek  indian  irish  italian  jamaican  japanese  \
0               7      40      6      34      1      128         2         2   
1               1       0      0       1      1        1         1         0   
2               0       1      2       4      0        2         0        11   
3               0       1      0       0      0       67         0         0   
4               1      38     32      24     20       12       195         1   
5               2     176     29     124     13      266         3        13   
6               1      44      3       1      7       56         1         2   
7               0      13      0       4      0       15         1         1   
8               0       3      0       8      0        6         0         1   
9               0       0      0      69      0        0         0        17   
10              1       0      0       0      0        5         0         0   
11   

In [165]:
#calculate percentages

In [166]:
pct_counts = raw_counts.copy()

In [167]:
#change ingredient counts to ingredient percentages
for i, row in pct_counts.iterrows():
    #print(i, row['ingredient'])
    for key, val in cuisine_counts.items():
        #key - the cuisine
        #val - the total number of recipes of that cuisine
        #row[key] - the number of recipes of that cuisine with the ingredient
        #row[key] / val - perecentage of recipes with the ingredient 
        #print (key, val, row[key], row[key]/val)  
        percentage = row[key]/val
        pct_counts.loc[i, key] = percentage

In [168]:
pct_counts.shape

(915, 21)

In [169]:
pct_counts.head(20)

cuisine          ingredient  brazilian   british  cajun_creole   chinese  \
0              active yeast   0.006424  0.016169      0.065934  0.004489   
1               adobo sauce   0.000000  0.000000      0.000000  0.000000   
2              agave nectar   0.008565  0.000000      0.000000  0.001496   
3             alfredo sauce   0.000000  0.000000      0.009158  0.000000   
4                  allspice   0.002141  0.033582      0.032967  0.000000   
5                    almond   0.051392  0.054726      0.021978  0.017209   
6            almond extract   0.000000  0.013682      0.003663  0.002993   
7              almond flour   0.000000  0.001244      0.001832  0.001871   
8               almond milk   0.027837  0.001244      0.000000  0.000000   
9                    amchur   0.000000  0.000000      0.000000  0.000000   
10          american cheese   0.000000  0.000000      0.003663  0.000000   
11            anaheim chile   0.000000  0.000000      0.000000  0.000000   
12                    ancho   0.002141  0.000000      0.003663  0.000000   
13       ancho chile pepper   0.000000  0.000000      0.000000  0.000000   
14                  anchovy   0.000000  0.001244      0.009158  0.000000   
15            anchovy paste   0.000000  0.000000      0.005495  0.000000   
16        andouille sausage   0.002141  0.000000      0.417582  0.000000   
17               angel hair   0.000000  0.000000      0.007326  0.001496   
18                    anise   0.004283  0.001244      0.000000  0.041526   
19               anise seed   0.002141  0.000000      0.000000  0.000374   

cuisine  filipino    french     greek    indian     irish   italian  jamaican  \
0        0.009272  0.015117  0.005106  0.011322  0.001499  0.016331  0.003802   
1        0.001325  0.000000  0.000000  0.000333  0.001499  0.000128  0.001901   
2        0.000000  0.000378  0.001702  0.001332  0.000000  0.000255  0.000000   
3        0.000000  0.000378  0.000000  0.000000  0.000000  0.008548  0.000000   
4        0.001325  0.014361  0.027234  0.007992  0.029985  0.001531  0.370722   
5        0.002649  0.066515  0.024681  0.041292  0.019490  0.033937  0.005703   
6        0.001325  0.016629  0.002553  0.000333  0.010495  0.007145  0.001901   
7        0.000000  0.004913  0.000000  0.001332  0.000000  0.001914  0.001901   
8        0.000000  0.001134  0.000000  0.002664  0.000000  0.000766  0.000000   
9        0.000000  0.000000  0.000000  0.022977  0.000000  0.000000  0.000000   
10       0.001325  0.000000  0.000000  0.000000  0.000000  0.000638  0.000000   
11       0.001325  0.000000  0.000000  0.000333  0.000000  0.000000  0.000000   
12       0.000000  0.000378  0.000000  0.000333  0.000000  0.000128  0.000000   
13       0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   
14       0.000000  0.026833  0.004255  0.000000  0.000000  0.025772  0.000000   
15       0.000000  0.004157  0.001702  0.000000  0.000000  0.005996  0.000000   
16       0.000000  0.000000  0.000000  0.000000  0.000000  0.000255  0.000000   
17       0.002649  0.000756  0.003404  0.000000  0.000000  0.008548  0.001901   
18       0.017219  0.006047  0.008511  0.017316  0.004498  0.005231  0.005703   
19       0.000000  0.001512  0.003404  0.000999  0.001499  0.001659  0.001901   

cuisine  japanese    korean   mexican  moroccan   russian  southern_us  \
0        0.001405  0.000000  0.002641  0.009744  0.071575     0.006713   
1        0.000000  0.000000  0.013203  0.000000  0.000000     0.000926   
2        0.007730  0.007229  0.004349  0.000000  0.004090     0.001620   
3        0.000000  0.000000  0.000155  0.000000  0.000000     0.000000   
4        0.000703  0.000000  0.010252  0.053593  0.016360     0.013657   
5        0.009136  0.001205  0.012892  0.135201  0.053170     0.016204   
6        0.001405  0.000000  0.000932  0.001218  0.010225     0.005324   
7        0.000703  0.000000  0.000621  0.000000  0.000000     0.002315   
8        0.000703  0.000000  0.0004

In [170]:
prevalence_ratios = pct_counts.copy()

now compute overall percentages for each ingredient

In [171]:
#THIS IS WRONG... computing an average accross the entire 
#recipe dataset overweights the more prevalent cuisines
#now compute overall percentages for each ingredient
# ingredient_pcts = {}
# for ing in ingredient_counts.columns:
#     if ing == 'cuisine':
#         pass
#     else:
#         ing_count = ingredient_counts[ing].sum()
#         ing_pct = ing_count/recipe_total
#         #print (ing, ing_count, ing_pct )
#         ingredient_pcts[ing] = ing_pct

# for i, row in ic2.iterrows():
#     #print(i, row['ingredient'])
#     for key, val in cuisine_counts.items():
#         #key - the cuisine
#         #row[key] - percentage of recipes in that cuisione with that ingredient
#         #ingredient_pcts[row['ingredient]] - percentage across all cuisines 
#         if ingredient_pcts[row['ingredient']] == 0:
#             relative_ratio = 0
#         else:    
#            relative_ratio = row[key] / ingredient_pcts[row['ingredient']]
#         #print (key, row[key], ingredient_pcts[row['ingredient']], relative_ratio) 
#         ic2.loc[i, key] = relative_ratio

In [172]:
prevalence_ratios['avg'] = np.nan
for i, row in prevalence_ratios.iterrows():
    #print (row.ingredient, row[1:21].mean())
    avg = row[1:21].mean()
    prevalence_ratios.loc[i, 'avg'] = avg
    for cuisine in cuisine_counts.keys():
        #print (cuisine, row[cuisine], avg, row[cuisine] / avg)
        if avg == 0:
            relative_ratio = 0
        else:    
            relative_ratio = row[cuisine] / avg
        prevalence_ratios.loc[i, cuisine] = relative_ratio

        

In [173]:
prevalence_ratios.head(40)

cuisine          ingredient  brazilian   british  cajun_creole   chinese  \
0              active yeast   0.506923  1.275924      5.202923  0.354258   
1               adobo sauce   0.000000  0.000000      0.000000  0.000000   
2              agave nectar   3.407026  0.000000      0.000000  0.595242   
3             alfredo sauce   0.000000  0.000000     10.041753  0.000000   
4                  allspice   0.068419  1.072999      1.053347  0.000000   
5                    almond   1.528444  1.627616      0.653648  0.511816   
6            almond extract   0.000000  3.343319      0.895114  0.731362   
7              almond flour   0.000000  1.056741      1.556080  1.589263   
8               almond milk  15.054660  0.672649      0.000000  0.000000   
9                    amchur   0.000000  0.000000      0.000000  0.000000   
10          american cheese   0.000000  0.000000      6.817935  0.000000   
11            anaheim chile   0.000000  0.000000      0.000000  0.000000   
12                    ancho   0.734606  0.000000      1.256634  0.000000   
13       ancho chile pepper   0.000000  0.000000      0.000000  0.000000   
14                  anchovy   0.000000  0.231060      1.701211  0.000000   
15            anchovy paste   0.000000  0.000000      5.648447  0.000000   
16        andouille sausage   0.098522  0.000000     19.212900  0.000000   
17               angel hair   0.000000  0.000000      4.207221  0.859387   
18                    anise   0.375155  0.108954      0.000000  3.637655   
19               anise seed   1.724780  0.000000      0.000000  0.301337   
20                    apple   0.849587  1.480438      1.089993  0.385922   
21              apple cider   0.280618  0.162996      1.440097  1.127618   
22              apple juice   1.123862  0.000000      0.000000  0.196350   
23                  apricot   0.192756  1.567465      0.000000  0.134706   
24             arborio rice   0.000000  0.000000      1.738524  0.118373   
25                artichoke   0.000000  0.236470      3.482091  0.000000   
26          artichoke heart   0.000000  0.000000      2.245807  0.000000   
27                  arugula   0.000000  1.124197      1.103607  0.112714   
28                asafetida   0.000000  0.000000      0.000000  0.000000   
29               asafoetida   0.000000  0.000000      0.000000  0.000000   
30                   asiago   0.000000  0.000000      2.209964  0.000000   
31                 asparagu   0.000000  0.190736      1.685180  1.147410   
32           asparagu spear   0.000000  0.000000      1.357902  0.832115   
33                  avocado   1.273318  0.000000      0.435634  0.200215   
34                    bacon   2.109517  1.021086      3.728873  0.319413   
35           bacon dripping   0.000000  1.576648      8.125800  0.000000   
36                 baguette   0.539529  0.000000      2.538058  0.047131   
37                   baking   0.469020  3.068401      1.309041  0.431276   
38            baking potato   0.000000  2.223812      1.007577  0.051453   
39              baking soda   0.370150  3.138997      1.456328  0.594953   

cuisine  filipino    french     greek     indian     irish    italian  \
0        0.731625  1.192910  0.402950   0.893431  0.118307   1.288672   
1        1.371529  0.000000  0.000000   0.344823  1.552480   0.132113   
2        0.000000  0.150329  0.677056   0.529830  0.000000   0.101498   
3        0.000000  0.414422  0.000000   0.000000  0.000000   9.373499   
4        0.042320  0.458865  0.870169   0.255357  0.958066   0.048918   
5        0.078784  1.978236  0.734033   1.228066  0.579659   1.009326   
6        0.323664  4.063535  0.623914   0.081374  2.564563   1.745918   
7        0.000000  4.174247  0.000000   1.131695  0.000000   1.625963   
8        0.000000  0.613163  0.000000   1.440718  0.000000   0.413991   
9        0.000000  0.000000  0.000000  13.071802  0.000000   0.000000   
10       2.465293  0.000000  0.000000   0.000000  0.000000   1.187354   
11     

In [174]:
for cuisine in cuisine_counts.keys():
    print(cuisine)

brazilian
british
cajun_creole
chinese
filipino
french
greek
indian
irish
italian
jamaican
japanese
korean
mexican
moroccan
russian
southern_us
spanish
thai
vietnamese


In [175]:
prevalence_ratios[['ingredient','brazilian']].sort_values('brazilian',ascending=False).head(10)

cuisine       ingredient  brazilian
114              cachaca  20.000000
8            almond milk  15.054660
834        tapioca flour  13.033896
799             sprinkle  10.169709
771         simple syrup   9.810379
835       tapioca starch   9.129978
219       condensed milk   8.412668
814      superfine sugar   8.147710
434             key lime   7.094462
72            black bean   6.633678

In [176]:
prevalence_ratios[['ingredient','british']].sort_values('british',ascending=False).head(10)

cuisine      ingredient    british
353        golden syrup  16.660127
276        double cream  12.331458
519         mixed spice  10.609049
585         pastry puff   8.512946
264             currant   8.095681
701          rolled oat   8.034964
665         puff pastry   8.007904
478        malt vinegar   7.455985
355      graham cracker   7.372087
358       grand marnier   6.837118

In [177]:
prevalence_ratios[['ingredient','cajun_creole']].sort_values('cajun_creole',ascending=False).head(10)

cuisine         ingredient  cajun_creole
308                   file     19.865494
115        cajun seasoning     19.338700
254       creole seasoning     19.268229
16       andouille sausage     19.212900
253         creole mustard     18.607071
247               crawfish     18.104390
780         smoked sausage     17.880724
546                   okra     14.986103
130         catfish fillet     14.870321
745              seasoning     14.139487

In [178]:
prevalence_ratios[['ingredient','chinese']].sort_values('chinese',ascending=False).head(10)

cuisine           ingredient    chinese
303          fermented black  20.000000
825      szechwan peppercorn  17.932449
189                chow mein  17.310654
758            shaoxing wine  16.833215
81             black vinegar  16.175565
316               five spice  13.381917
480          mandarin orange  11.621275
282         dumpling wrapper  11.106715
907           wonton wrapper  11.069160
893           water chestnut  10.843581

In [179]:
prevalence_ratios[['ingredient','filipino']].sort_values('filipino',ascending=False).head(10)

cuisine       ingredient   filipino
140              chayote  13.045494
347        glutinou rice  10.084641
640           pork belly   8.821861
769         shrimp paste   8.747703
619      pineapple chunk   7.628808
59            beef shank   7.212414
654        powdered milk   6.795297
700         roll wrapper   6.586729
206        coconut cream   6.158839
641            pork butt   5.731952

In [180]:
prevalence_ratios[['ingredient','french']].sort_values('french',ascending=False).head(10)

cuisine      ingredient     french
385             gruyère  16.002614
384      gruyere cheese  15.300329
99          brie cheese  14.860179
270         de provence  10.964360
215              cognac  10.666840
393        haricot vert  10.211675
823         swis cheese   9.373758
358       grand marnier   9.348713
836            tarragon   9.273547
593         pearl onion   8.834360

In [181]:
prevalence_ratios[['ingredient','greek']].sort_values('greek',ascending=False).head(10)

cuisine       ingredient      greek
368      greek seasoning  19.403674
304          feta cheese  17.631974
624                 pita  16.310716
625           pita bread  16.131310
607         phyllo dough  13.099178
429       kalamata olive  12.768651
428             kalamata  12.116811
561                 orzo  11.598318
369         greek yogurt  11.146029
514           mint flake  10.982036

In [182]:
prevalence_ratios[['ingredient','indian']].sort_values('indian',ascending=False).head(10)

cuisine      ingredient     indian
868            urad dal  18.847139
75        black mustard  18.284583
29           asafoetida  17.752217
357          gram flour  17.603304
569              paneer  17.296577
333        garam masala  17.022286
490              masala  16.920502
302           fenugreek  16.856130
420             jaggery  16.734017
163      chickpea flour  16.262153

In [183]:
prevalence_ratios[['ingredient','irish']].sort_values('irish',ascending=False).head(10)

cuisine      ingredient      irish
412       irish whiskey  19.175450
233         corned beef  15.423777
56         beef brisket  10.896447
492       mashed potato  10.099121
592        pearl barley   8.943222
902             whiskey   8.759590
898         wheat flour   7.943178
469             liqueur   7.930646
701          rolled oat   6.779727
183      chocolate chip   6.624380

In [184]:
prevalence_ratios[['ingredient','italian']].sort_values('italian',ascending=False).head(10)

cuisine       ingredient    italian
348              gnocchi  20.000000
101        broccoli rabe  18.825495
486       marinara sauce  16.608556
582          pasta shell  16.167911
443       lasagna noodle  15.869883
606                pesto  14.878290
694             rigatoni  14.845468
853           tortellini  14.773372
596      pecorino romano  14.716509
693       ricotta cheese  14.557840

In [185]:
prevalence_ratios[['ingredient','jamaican']].sort_values('jamaican',ascending=False).head(10)

cuisine       ingredient   jamaican
741        scotch bonnet  18.662653
388      habanero pepper  16.854383
4               allspice  11.845144
710                  rum  11.440904
629             plantain   8.439005
848       tomato ketchup   8.249883
910                  yam   8.194589
68                berrie   7.090038
478         malt vinegar   6.964590
620      pineapple juice   6.662645

In [186]:
prevalence_ratios[['ingredient','japanese']].sort_values('japanese',ascending=False).head(10)

cuisine    ingredient   japanese
891      wasabi paste  19.884340
88       bonito flake  19.633264
437             konbu  19.218338
268             dashi  18.903106
890            wasabi  18.660672
335              gari  17.102154
716              sake  16.896478
516              miso  16.744592
539              nori  16.340665
517        miso paste  15.919718

In [187]:
prevalence_ratios[['ingredient','korean']].sort_values('korean',ascending=False).head(10)

cuisine       ingredient     korean
350            gochugaru  20.000000
436               kimchi  19.651063
351            gochujang  19.521585
683            rice cake  18.193235
44                  base  15.198773
591                 pear  12.063208
755               sesame   8.761346
756           sesame oil   8.008261
294            enokitake   7.856592
565      oyster mushroom   7.782597

In [188]:
prevalence_ratios[['ingredient','mexican']].sort_values('mexican',ascending=False).head(10)

cuisine         ingredient    mexican
499           mexican beer  20.000000
501      mexican chocolate  20.000000
827             taco sauce  20.000000
292        enchilada sauce  19.925763
679           refried bean  19.553387
828         taco seasoning  19.545606
503        mexican oregano  19.368108
725            salsa verde  19.112867
579          pasilla chile  19.053500
489            masa harina  18.830709

In [189]:
prevalence_ratios[['ingredient','moroccan']].sort_values('moroccan',ascending=False).head(10)

cuisine       ingredient   moroccan
660      preserved lemon  19.443454
394              harissa  19.177320
241              couscou  17.541391
322         flower water  13.417313
664                prune  11.125835
162             chickpea  10.437182
23               apricot  10.416125
441        lamb shoulder   9.577349
440           lamb shank   9.106026
448             leg lamb   8.815954

In [190]:
prevalence_ratios[['ingredient','russian']].sort_values('russian',ascending=False).head(10)

cuisine      ingredient    russian
65                 beet  14.825183
136         celery root  10.857059
273         dill pickle  10.314269
238      cottage cheese  10.210108
274            dillweed   9.710540
272                dill   9.703913
636               poppy   9.269208
779       smoked salmon   7.896720
124             caraway   7.517056
235           cornichon   7.050734

In [191]:
prevalence_ratios[['ingredient','southern_us']].sort_values('southern_us',ascending=False).head(10)

cuisine        ingredient  southern_us
240           country ham    17.959393
512      mini marshmallow    14.649829
873         vanilla wafer    14.149470
614             pie shell    12.544980
90        bourbon whiskey    12.517970
383                  grit    11.738923
382          green tomato    10.773335
587                peache    10.729411
185            chop pecan    10.561533
434              key lime     9.970018

In [192]:
prevalence_ratios[['ingredient','spanish']].sort_values('spanish',ascending=False).head(10)

cuisine       ingredient    spanish
791      spanish chorizo  17.738137
479      manchego cheese  16.899289
793      spanish paprika  16.361773
188      chorizo sausage  12.378269
763       sherry vinegar  12.305277
187              chorizo  12.239623
472      littleneck clam   9.513130
239        country bread   9.449654
397      heirloom tomato   9.240511
856           triple sec   8.927872

In [193]:
prevalence_ratios[['ingredient','thai']].sort_values('thai',ascending=False).head(10)

cuisine      ingredient       thai
375         green curry  19.000279
332            galangal  18.608400
266         curry paste  17.182031
807      straw mushroom  15.919292
427         kaffir lime  15.906822
567          palm sugar  14.660376
833      tamarind paste  12.108833
589       peanut butter  11.967106
450          lemon gras  11.460687
457           lemongras  10.091535

In [194]:
prevalence_ratios[['ingredient','vietnamese']].sort_values('vietnamese',ascending=False).head(10)

cuisine       ingredient  vietnamese
686           rice paper   16.165381
688      rice vermicelli   15.025810
697           rock sugar   12.740472
882           vermicelli   12.350494
908             wood ear   11.445955
69            bird chile   10.207015
685          rice noodle    8.677374
314           fish sauce    8.649910
803           star anise    8.545311
52            beansprout    8.523703

# Similarity of cuisines

Try using cosine similarity to find most similar cuisines

In [195]:
from sklearn.metrics.pairwise import cosine_similarity

In [196]:
pr2 = prevalence_ratios.T.reset_index().iloc[1:]
pr2 = pr2.set_index('cuisine')
pr2 = pr2.iloc[0:20,:]

In [197]:
#pr2

In [198]:
cuisine_sim = pd.DataFrame(
    cosine_similarity(pr2),
    columns=pr2.index,
    index=pr2.index
)

In [199]:
cuisine_sim

cuisine       brazilian   british  cajun_creole   chinese  filipino    french  \
cuisine                                                                         
brazilian      1.000000  0.192706      0.200489  0.106572  0.215039  0.178016   
british        0.192706  1.000000      0.212391  0.109229  0.153778  0.425554   
cajun_creole   0.200489  0.212391      1.000000  0.125856  0.207012  0.257373   
chinese        0.106572  0.109229      0.125856  1.000000  0.337562  0.111570   
filipino       0.215039  0.153778      0.207012  0.337562  1.000000  0.144930   
french         0.178016  0.425554      0.257373  0.111570  0.144930  1.000000   
greek          0.070197  0.131845      0.179607  0.062149  0.091171  0.210034   
indian         0.100748  0.121560      0.081193  0.103551  0.104203  0.075250   
irish          0.179683  0.436757      0.205890  0.087901  0.137779  0.306798   
italian        0.096170  0.170062      0.258676  0.065565  0.095606  0.298912   
jamaican       0.243711  0.228213      0.222319  0.162103  0.251716  0.148132   
japanese       0.075414  0.087590      0.086341  0.240425  0.140358  0.093030   
korean         0.093975  0.081827      0.094362  0.298195  0.209632  0.089134   
mexican        0.129335  0.057058      0.169694  0.057137  0.110793  0.069984   
moroccan       0.100608  0.178349      0.135684  0.079855  0.098647  0.193745   
russian        0.177645  0.346271      0.218055  0.110180  0.153636  0.343743   
southern_us    0.225247  0.331353      0.355315  0.099965  0.175599  0.250593   
spanish        0.208447  0.185021      0.251451  0.084497  0.140515  0.319836   
thai           0.161978  0.070075      0.121436  0.309015  0.299085  0.097383   
vietnamese     0.151362  0.084465      0.136800  0.404357  0.346230  0.119206   

cuisine          greek    indian     irish   italian  jamaican  japanese  \
cuisine                                                                    
brazilian     0.070197  0.100748  0.179683  0.096170  0.243711  0.075414   
british       0.131845  0.121560  0.436757  0.170062  0.228213  0.087590   
cajun_creole  0.179607  0.081193  0.205890  0.258676  0.222319  0.086341   
chinese       0.062149  0.103551  0.087901  0.065565  0.162103  0.240425   
filipino      0.091171  0.104203  0.137779  0.095606  0.251716  0.140358   
french        0.210034  0.075250  0.306798  0.298912  0.148132  0.093030   
greek         1.000000  0.106536  0.129090  0.283056  0.100395  0.066250   
indian        0.106536  1.000000  0.093343  0.040335  0.146529  0.178675   
irish         0.129090  0.093343  1.000000  0.140011  0.166332  0.072130   
italian       0.283056  0.040335  0.140011  1.000000  0.091411  0.052651   
jamaican      0.100395  0.146529  0.166332  0.091411  1.000000  0.084305   
japanese      0.066250  0.178675  0.072130  0.052651  0.084305  1.000000   
korean        0.066075  0.053294  0.083514  0.047551  0.104156  0.292267   
mexican       0.079985  0.045090  0.063412  0.076348  0.110038  0.033406   
moroccan      0.294145  0.196128  0.157464  0.120003  0.145522  0.076012   
russian       0.195103  0.113306  0.338999  0.158716  0.158368  0.117713   
southern_us   0.122775  0.068210  0.307575  0.143780  0.210894  0.065020   
spanish       0.194687  0.085324  0.157185  0.232126  0.144822  0.067833   
thai          0.087598  0.176227  0.061001  0.070546  0.178210  0.174209   
vietnamese    0.095892  0.115502  0.081056  0.059205  0.148908  0.177402   

cuisine         korean   mexican  moroccan   russian  southern_us   spanish  \
cuisine                                                                       
brazilian     0.093975  0.129335  0.100608  0.177645     0.225247  0.208447   
british       0.081827  0.057058  0.178349  0.346271     0.331353  0.185021   
cajun_creole  0.094362  0.169694  0.135684  0.218055     0.355315  0.251451   
chinese       0.298195  0.057137  0.079855  0.110180     0.099965  0.084497   
filipino      0.209632  0.110793  0.098647  0.153636     0.175599  0.1

In [200]:
for cuisine in cuisine_counts.keys():
    print(cuisine_sim[cuisine].sort_values(ascending=False).head(10))

cuisine
brazilian       1.000000
jamaican        0.243711
southern_us     0.225247
filipino        0.215039
spanish         0.208447
cajun_creole    0.200489
british         0.192706
irish           0.179683
french          0.178016
russian         0.177645
Name: brazilian, dtype: float64
cuisine
british         1.000000
irish           0.436757
french          0.425554
russian         0.346271
southern_us     0.331353
jamaican        0.228213
cajun_creole    0.212391
brazilian       0.192706
spanish         0.185021
moroccan        0.178349
Name: british, dtype: float64
cuisine
cajun_creole    1.000000
southern_us     0.355315
italian         0.258676
french          0.257373
spanish         0.251451
jamaican        0.222319
russian         0.218055
british         0.212391
filipino        0.207012
irish           0.205890
Name: cajun_creole, dtype: float64
cuisine
chinese         1.000000
vietnamese      0.404357
filipino        0.337562
thai            0.309015
korean          0.298

Most similar cuisines:

thai - vietnamese: .51

british - irish: .44

french - british: .43

chinese - vietnamese: .40

southern us - cajun-creole: .36

In [201]:
for cuisine in cuisine_counts.keys():
    print(cuisine_sim[cuisine].sort_values().head(1))

cuisine
greek    0.070197
Name: brazilian, dtype: float64
cuisine
mexican    0.057058
Name: british, dtype: float64
cuisine
indian    0.081193
Name: cajun_creole, dtype: float64
cuisine
mexican    0.057137
Name: chinese, dtype: float64
cuisine
greek    0.091171
Name: filipino, dtype: float64
cuisine
mexican    0.069984
Name: french, dtype: float64
cuisine
chinese    0.062149
Name: greek, dtype: float64
cuisine
italian    0.040335
Name: indian, dtype: float64
cuisine
thai    0.061001
Name: irish, dtype: float64
cuisine
indian    0.040335
Name: italian, dtype: float64
cuisine
japanese    0.084305
Name: jamaican, dtype: float64
cuisine
mexican    0.033406
Name: japanese, dtype: float64
cuisine
italian    0.047551
Name: korean, dtype: float64
cuisine
japanese    0.033406
Name: mexican, dtype: float64
cuisine
korean    0.067626
Name: moroccan, dtype: float64
cuisine
mexican    0.067983
Name: russian, dtype: float64
cuisine
japanese    0.06502
Name: southern_us, dtype: float64
cuisine
japane

Least similar cuisines:

mexican - japanese: .033

italian - indian: .040

korean - italian: .047

british - mexican: .057
